app/migration/migrators/base.rb

```diff
@@ -160,5 +168,13 @@ def finalise_migration!
       # dependent models.
       MigrationJob.set(wait: 10.seconds).perform_later
     end
+
+    def lead_provider_ids_by_name
+      @lead_provider_ids_by_name ||= ::LeadProvider.pluck(:name, :id).to_h
+    end
+
+    def active_lead_provider_ids_by_lead_provider_and_registration_period
+      @active_lead_provider_ids_by_lead_provider_and_registration_period ||= ::ActiveLeadProvider.pluck(:lead_provider_id, :registration_period_id, :id).to_h { |s| ["#{s[0]} #{s[1]}", s[2]] }
```

Comment:
There's no need to join learn provider id and registration period into a string, you can keep them as integer tuple:
```ruby
@active_lead_provider_ids_by_lead_provider_and_registration_period ||= ::ActiveLeadProvider.pluck(:lead_provider_id, :registration_period_id, :id).to_h do |s|
  [
    [s[0], s[1]],
    s[2]
  ]
end
```